tests/catalog/test_hive.py

```diff
@@ -294,6 +294,37 @@ def test_create_table(table_schema_simple: Schema, hive_database: HiveDatabase,
 
     assert metadata.model_dump() == expected.model_dump()
 
+    catalog.create_table(
+        ("default", "table_v1"), schema=table_schema_simple, properties={"owner": "javaberg", "format-version": "1"}
+    )
+    expected_v1 = TableMetadataV1(
+        location=metadata.location,
+        table_uuid=metadata.table_uuid,
+        last_updated_ms=metadata.last_updated_ms,
+        last_column_id=3,
+        schema=Schema(
```

Comment:
I know that it is still required, but also deprecated:
<img width="707" alt="image" src="https://github.com/apache/iceberg-python/assets/1134248/3b96e21c-9d41-40cb-be87-e583d166341d">

Setting `schemas` makes more sense to me.

Reply:
Yeah I was thinking about this, since it's required do we want to have a validator which sets the Schema based on current-schema-id and schemas? Then when the V1 metadata gets written out, all 3 fields will be set.

Reply:
Updated